app/lib/katello/logging.rb

```diff
@@ -9,6 +9,33 @@ def self.time(message, data: {}, logger: Rails.logger, level: :info)
       data_string = data.map { |k, v| "#{k}=#{v}" }.join(' ')
 
       logger.send(level, "#{message} #{data_string}")
+      "#{message} #{data_string}"
+    end
+
+    class Timer
+      def initialize(key = "default")
+        @key = key
+        @start_time = Time.now
+        @@timers ||= {}
+        @@timers[key] = self
```

Comment:
I'm not sure how thread safe this is given that its using class variables.  I wonder if we should use Thread local varaibles?  (Thread.current[:key]) ?